src/pgwire/src/protocol.rs

```diff
@@ -1076,6 +1080,22 @@ where
                     );
                     self.conn.send(msg).await?;
                 }
+
+                // Make sure any writes have been persisted as necessary.
+                if let Some(mut rx) = rx {
+                    loop {
+                        match rx.recv().await {
```

Comment:
As I stated above, I think this is the wrong design. Should the presence of an error change the tag? Did the transaction partially commit? Are there atomicity guarantees provided or not provided by persistence? If persistence isn't atomic then we may have to rethink it or what transactions allow.